app/controllers/posts_controller.rb

```diff
@@ -2,7 +2,7 @@ class PostsController < ApplicationController
   before_action :fetch_author
 
   def index
-    @posts = @user.posts
+    @posts = Post.all
   end
```

Comment:
- [ ] In demonstrating the elimination of N+1 query problem, you need to consider that we want to fetch all posts, but we also what to fetch their respective comments along with them. Kindly modify your index action here to do that. See example.

  ```
  @posts = @user.posts.includes(:comments)
  ```